### ISSUE TITLE
Need way to specify tip date trait value format in BEAUti

### ISSUE THREAD

**Issue by tgvaughan** [CompEvol/beast2#729]
TraitSet allows users to specify the format that dates come in, but this useful functionality is not exposed to BEAUti users.

A related point is that the TipDatesInput editor uses its own code to do the parsing of dates, duplicating existing code in TraitSet.

**Comment by tgvaughan** [comment]
I've just pushed up a [draft implementation](https://github.com/tgvaughan/beast2/commit/15fc235236f7e65643c9b314b81f9c98b1ed896a) to my fork.  The UI needs some work (the button box is now a bit too wide) but here's what it looks like:
![screen shot 2017-08-30 at 12 02 10](https://user-images.githubusercontent.com/512538/29870481-f5dbf2d8-8d87-11e7-899a-40681940779f.png)

This implementation actually required a bit of messing arround/refactoring.  I've summarized what went into it in the commit message, but I've pasted the points below so you don't have to go hunting:

This commit does a few things:
1. Causes the "Date" column of the tip date input editor table to
contain the raw trait values.
2. Stores these raw values in the corresponding `TraitSet`. This means
that the XML file contains the original trait values (eg. dates like
2014-03-13).
3. Uses a previously private method in the `TraitSet` class to process
the dates (this is the same method that BEAST uses) and populate the
"Height" column of the input editor table.
4. Moves the error handling (previously a System.exit(1)) out of the
`TraitSet` trait value conversion method to an exception handler in the
calling code.
5. Provides a combo-box for specifying the date format that is recorded
in the dateFormat input to `TraitSet`.

As a happy side effect, badly-formatted tip date files (or badly specified formats) no longer break the input editor (the example problem described in #728).

Any thoughts?
